### ISSUE TITLE
Add examples on how to use the Extras

### ISSUE THREAD

**Issue by benthecarman** [Kixunil/bip21#11]
For now can look at @DanGould's 

https://github.com/chaincase-app/payjoin/blob/master/payjoin/src/uri.rs

**Comment by DanGould** [comment]
That's all @Kixunil's work, not mine. Still useful to have examples, especially when this crate gets used in bindings

**Comment by DanGould** [comment]
I think we could show how library-defined `Extras` could be composed. For example, `bitcoin-walia` duplicates `payjoin` bip21 parsing by defining `WaliaExtras` as follows:

https://github.com/MutinyWallet/bitcoin-waila/blob/627375a3b628616ccea6d65bd8bc912558fddcf9/waila/src/bip21.rs#L15-L19

and later duplicating the de/serialization logic

Where it could be defined as 

```rs
pub struct WailaExtras {
    pub lightning: Option<Bolt11Invoice>,
    pub pj_extras: Option<PayjoinExtras>
}
```

and then use the bip21 defined functions like `PayjoinExtras::deserialize_temp` within its own
`WaliaExtras::deserialize_temp`

**Comment by Kixunil** [comment]
Good point. I think I had some macros in mind to help with this but I don't remember it well.

**Comment by DanGould** [comment]
Macros would be great to allow builders for `Uri<Extras>`
